lib/ziti-tunnel-cbs/ziti_dns.c

```diff
@@ -113,7 +114,29 @@ struct ziti_dns_s {
 } ziti_dns;
 
 static uint32_t next_ipv4() {
-   return  htonl(ziti_dns.ip_pool.base | (ziti_dns.ip_pool.counter++ & ziti_dns.ip_pool.counter_mask));
+    uint32_t candidate;
+    uint32_t i = 0; // track how many candidates have been considered. should never exceed pool capacity.
```

Comment:
The current logic is based on scanning the entire map, because `ziti_dns.ip_addresses` will be fragmented over time. I think we could miss available IPs if we don't check the full range?

The counter in the loop is a fail-safe to avoid looping forever if there is an unforeseen programming error. The test above the loop _should_ prevent us from entering the loop when the scan is doomed.

Reply:
> would this be easier if we start with capacity - model_map_size(addresses) and just found the next one below?

the ip selection currently depends on incrementing `ip_pool.counter`, and once the counter wraps we don't know if the "next one" is available until we look for it.